Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -33,6 +32,88 @@ void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
+    amrex::Real const coef2_r = 2._rt*cdt_over_dr/(1._rt + cdt_over_dr) / PhysConst::c;
+    amrex::Real const coef3_r = cdt/(1._rt + cdt_over_dr) / PhysConst::c;
+
+    // Extract stencil coefficients
+    Real const * const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+    int const n_coefs_z = m_stencil_coefs_z.size();
+
+    // Extract cylindrical specific parameters
+    Real const dr = m_dr;
+    int const nmodes = m_nmodes;
+    Real const rmin = m_rmin;
+
+    // tiling is usually set by TilingIfNotGPU()
+    // but here, we set it to false because of potential race condition,
+    // since we grow the tiles by one guard cell after creating them.
+    for ( MFIter mfi(*Efield[0], false); mfi.isValid(); ++mfi ) {
+        // Extract field data for this grid/tile
+        Array4<Real> const& Er = Efield[0]->array(mfi);
+        Array4<Real> const& Et = Efield[1]->array(mfi);
+        Array4<Real> const& Ez = Efield[2]->array(mfi);
+        // Array4<Real> const& Br = Bfield[0]->array(mfi);
+        Array4<Real> const& Bt = Bfield[1]->array(mfi);
+        Array4<Real> const& Bz = Bfield[2]->array(mfi);
+
+        // Extract tileboxes for which to loop
+        Box tbr  = mfi.tilebox(Bfield[0]->ixType().toIntVect());
+        Box tbt  = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+        Box tbz  = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+        // We will modify the first (i.e. innermost) guard cell
+        // (if it is outside of the physical domain)
+        // Thus, the tileboxes here are grown by 1 guard cell
+        tbr.grow(1);
+        tbt.grow(1);
+        tbz.grow(1);
+
+        // Loop over the cells
+        amrex::ParallelFor(tbr, tbt, tbz,
+            [=] AMREX_GPU_DEVICE (int /*i*/, int /*j*/, int /*k*/){},
+            [=] AMREX_GPU_DEVICE (int i, int j, int /*k*/){
+
+                // At the +r boundary (innermost guard cell)
+                if ( i==domain_box.bigEnd(0)+1 ){
+                    // Mode 0
+                    Bt(i,j,0,0) = coef1_r*Bt(i,j,0,0) - coef2_r*Ez(i,j,0,0)
+                        + coef3_r*CylindricalYeeAlgorithm::UpwardDz(Er, coefs_z, n_coefs_z, i, j, 0, 0);
+                    for (int m=1; m<nmodes; m++) { // Higher-order modes
+                        // Real part
+                        Bt(i,j,0,2*m-1) = coef1_r*Bt(i,j,0,2*m-1) - coef2_r*Ez(i,j,0,2*m-1)
+                            + coef3_r*CylindricalYeeAlgorithm::UpwardDz(Er, coefs_z, n_coefs_z, i, j, 0, 2*m-1);
+                        // Imaginary part
+                        Bt(i,j,0,2*m) = coef1_r*Bt(i,j,0,2*m) - coef2_r*Ez(i,j,0,2*m)
+                            + coef3_r*CylindricalYeeAlgorithm::UpwardDz(Er, coefs_z, n_coefs_z, i, j, 0, 2*m);
+                    }
+                }
+            },
+            [=] AMREX_GPU_DEVICE (int i, int j, int /*k*/){
+
+                Real const r = rmin + i*dr; // r on nodal point (Br is nodal in r)
```

Comment:
Should this be modified because we are looking at a point for Bz?
```suggestion
                Real const r = rmin + (i+0.5_rt)*dr; // r on cell-centered point (Bz is cell-centered in r)
```

Reply:
Thank you for catching this Rémi! Of course, you are totally correct. I just fixed it.